packages/moonstone/Dialog/Dialog.js

```diff
@@ -185,28 +185,35 @@ const DialogBase = kind({
 
 	computed: {
 		className: ({noDivider, styler}) => styler.append({showDivider: !noDivider}),
+		generateId: () => () => Math.random().toString(36).substr(2, 8),
 		titleBelow: ({title, titleBelow}) => title ? titleBelow : ''
 	},
 
-	render: ({buttons, casing, css, children, title, titleBelow, ...rest}) => {
+	render: ({buttons, casing, css, children, generateId, title, titleBelow, ...rest}) => {
 		delete rest.noDivider;
 
+		const
+			buttonsId = generateId(),
```

Comment:
Do we have to regenerate on every render? If the purpose of having an ID is to sort the order of aria-label, can't we create ids per instances?

Reply:
@viodragon2 
I modified it not to generate every render.
But there is one thing to worry about.
It creates an id outside of kind, which is common to all dialogs.
If two or more dialogs appear on one screen, it may cause unexpected behavior.j

Reply:
This would be module-scoped and not instance-scoped. I suggest using `moonstone/internal/IdProvider` to resolve this. It's a HOC responsible for creating a unique id per component. You may extend the id to subcomponents by giving suffixes as `${id}_body`, `${id}_buttons`, etc.

Reply:
Thanks for your comment.
I updated using the IdProvider.